### DISCUSSION TITLE
How to use Vite and TypeScript in UI kit

### DISCUSSION THREAD

**Discussion by sonsu-lee** [quasarframework/quasar#16814] [Quasar App Extensions]
I'm using the Quasar ui kit. 

I'm currently using it with the default settings of webpack and javascript, and would like to migrate to vite and typescript. 

I was wondering if there is any documentation or discussion on this?

**Comment by dongwa** [comment]
https://quasar.dev/quasar-cli-vite/convert-to-quasar-cli-with-vite

**Reply by sonsu-lee** [reply]
`https://github.com/quasarframework/quasar/issues/7329` If this discussion is anything to go by, this is not yet supported by the UI-KIT APP EXTENSION, is it?

Or am I misunderstanding something?

**Comment by yusufkandemir** [comment]
Scaffolding for Vite/TS is not supported during `yarn create quasar`/`pnpm init quasar` yet. But, you can convert `@quasar/app-webpack` in `ui/package.json` and modify `ui/dev` to meet the Quasar CLI with Vite structure. You can also freely update your build scripts, which use Rollup, to suit your needs(e.g. adding TS support). For `app-extension`, you can directly convert your files to `.ts` if the target app uses TS, or introduce a build step to produce the same file structure in `dist` as js files, e.g. `app-extension/dist/index.js`.

I know it might be frustrating as there isn't an in-depth guide or scaffolding. But, this is how it is at the moment. We aim to update AE docs, scaffolding templates, etc. in the future to address this.
